lease/leasehttp/http.go

```diff
@@ -19,25 +19,37 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"time"
 
 	pb "github.com/coreos/etcd/etcdserver/etcdserverpb"
 	"github.com/coreos/etcd/lease"
 	"github.com/coreos/etcd/lease/leasepb"
 	"github.com/coreos/etcd/pkg/httputil"
+	"github.com/coreos/etcd/pkg/wait"
 	"golang.org/x/net/context"
 )
 
 var (
 	LeasePrefix         = "/leases"
 	LeaseInternalPrefix = "/leases/internal"
+	applyTimeout        = time.Second
 )
 
 // NewHandler returns an http Handler for lease renewals
-func NewHandler(l lease.Lessor) http.Handler {
-	return &leaseHandler{l}
+func NewHandler(l lease.Lessor, w wait.WaitTime, i Indexer) http.Handler {
```

Comment:
This is a bad interface; forces the handler to reason to know about indexes and WaitTime and introduces a lot of interfaces to pretend it's an appropriate abstraction. Having `func NewHandler(l lease.Lessor, waitch func() <-chan)` instead would be so much simpler; it can be called like `NewHandler(l, func() <-chan { return s.ApplyNotify()})`.